MapleServer2/PacketHandlers/Game/DungeonHandler.cs

```diff
@@ -57,14 +58,23 @@ public override void Handle(GameSession session, PacketReader packet)
         }
     }
 
+    //if player has both party dungeon session always join the solo session
+    //because the solo session would be -1 outside of the solo 
     public static void HandleEnterDungeonPortal(GameSession session)
     {
-        DungeonSession dungeonSession = GameServer.DungeonManager.GetDungeonSessionBySessionId(session.Player.DungeonSessionId);
-        if (dungeonSession == null)
-        {
-            return;
-        }
-        session.Player.Warp(dungeonSession.DungeonMapIds.First(), instanceId: dungeonSession.DungeonInstanceId);
+        //if player dungeon session is -1 they must be in a group dungeon, otherwise
+        //they would not be in a dungeon lobby where this function is called from
+        int dungeonSessionId = session.Player.DungeonSessionId == -1
```

Comment:
This should be a separate method, this is a long one-liner

Reply:
I changed the logic and made it a lot cleaner. 
I could not find a reason to use this ever again so I decided not to make a method for it. It's a unique case, but the calculation is simple now (yet difficult to understand, that's why I added a thorough comment).